src/main/java/hw02/Family.java

```diff
@@ -63,23 +63,31 @@ public void setPet(Pet pet) {
     }
 
     public boolean deleteChild(int index) {
-        for (int i = 0; i < children.length; i++) {
-            if (children[i].equals(children[index])) {
-                children[i] = null;
-                return true;
-            }
-        }
-        return false;
+        Human[] newChildren = new Human[children.length - 1];
+        System.arraycopy(children, 0, newChildren, 0, index);
+        System.arraycopy(children, index + 1, newChildren, index, children.length - index - 1);
```

Comment:
I think no need call this function twice, handle method like that call it at the end once